plumbing/object.go

```diff
@@ -12,6 +12,46 @@ var (
 	ErrInvalidType = errors.New("invalid object type")
 )
 
+// ObjectToPack is a representation of an object that is going to be into a
+// pack file. If it is a delta, Base is the object that this delta is based on
```

Comment:
Decide each field (Object, Base, Original, Depth) in their declaration, not in the common ObjectToPack docs.